adaptors/mqtt/test/e2e/usability/usability_test.go

```diff
@@ -0,0 +1,800 @@
+
+			By("then the two messages are the same", isMessagesTheSame)
+
+		})
+
+		Specify("subscribe messages through an MQTT device link whose payload is null", func() {
+
+			By("when deploy an MQTT device link with payload null", deployDeviceLinkWithPayloadNull)
+
+			By("then the device link is connected", isDeviceConnectedTrue)
+
+			By("when subscribe a message", subscribeMessage)
+
+			By("then the message is null", isSubscribeMessageNull)
+
+		})
+
+		Specify("if the MQTT server is unavailable", func() {
+
+			By("given the device link is connected", isDeviceConnectedTrue)
+
+			By("when invalid server URL", invalidServerURL)
+
+			By("then the device link is not connected", isDeviceConnectedFalse)
+
+			By("when correct server URL", correctServerURL)
+
+			By("then the device link is connected", isDeviceConnectedTrue)
+
+		})
+
+		Specify("subscribe messages through an MQTT device link whose payload is a complex JSON", func() {
+
+			By("when deploy an MQTT device link whose payload is a complex JSON", deployDeviceLinkWithComplexJSONValue)
+
+			By("then the device link is connected", isDeviceConnectedTrue)
+
+			By("when subscribe a message", subscribeMessage)
+
+			By("then the subscribed message is same to the JSON value", isSubscribedMessageValid)
+
+		})
+	})
+
+	Specify("if subscribe messages of templated topic", func() {
+
+		By("when deploy dummy device link with templated topic", deployDummyDeviceWithTemplatedTopic)
+
+		By("then the device link is connected", isDeviceConnectedTrue)
+
+		By("then succeed subscribing a message from topic", succeedSubscribingTemplatedTopic)
+	})
+
+	Context("operation on pods/CRD/nodes", func() {
+
+		Specify("if delete adaptor pods", func() {
+
+			By("given the device link is connected", isDeviceConnectedTrue)
+
+			By("when delete MQTT adaptor pods", deleteMQTTAdaptorPods)
+
+			By("then adaptor of the device link is not found", isAdaptorExistedFalse)
+
+		})
+
+		Specify("if delete limbs pods", func() {
+
+			By("given the device link is connected", isDeviceConnectedTrue)
+
+			By("when delete limbs pods", deleteLimbsPods)
+
+			By("then the MQTT adaptor pods become error", isMQTTAdaptorPodsError)
+
+		})
+
+		Specify("if delete MQTT device model", func() {
+
+			By("given the device link is connected", isDeviceConnectedTrue)
+
+			By("when delete MQTT device model", deleteMQTTDeviceModel)
+
+			By("then model of the device link is not found", isModelExistedFalse)
+
+			By("redeploy MQTT device model", redeployMQTTDeviceModel)
+
+		})
+
+		Specify("if delete corresponding node", func() {
+
+			By("given the device link is connected", isDeviceConnectedTrue)
+
+			By("when delete corresponding cluster node", deleteCorrespondingNode)
+
+			By("then node of the device link is not found", isNodeExistedFalse)
+
+			By("when recreate corresponding cluster node", createCorrespondingNode)
+
+			By("then node of the device link is found", isNodeExistedTrue)
+
+		})
+	})
+})
```

Comment:
Can you align this context to what you did in Dummy? And then recreating a Node is not just simply recreate the Node.

Reply:
updated